InstituteBudget.php

```diff
@@ -5,53 +5,98 @@
 require __DIR__ . '/vendor/autoload.php';
 class InstituteBudget extends \ExternalModules\AbstractExternalModule {
 	
+    //TODO Remove this constructor since external modules don't always play nice with them
 	public function __construct() {
 		parent::__construct();
 		
-		$pid = $this->getProjectId();
-		if (empty($pid)) {
-			return;
-		}
-		$budget_field = $this->getProjectSetting('budget_table_field');
-		$gng_field = $this->getProjectSetting('gonogo_table_field');
-		$summary_field = $this->getProjectSetting('summary_review_field');
 		
 		global $Proj;
 		if (gettype($Proj) == 'object') {
 			$this->proj = $Proj;
-			// collect instruments that hold schedule, gng, summary fields
-			foreach ($Proj->metadata as $field) {
-				if ($field['field_name'] === $budget_field) {
-					$this->budget_table_instrument = $field['form_name'];
-				}
-				if ($field['field_name'] === $gng_field) {
-					$this->gonogo_table_instrument = $field['form_name'];
-				}
-				if ($field['field_name'] === $summary_field) {
-					$this->summary_review_instrument = $field['form_name'];
-				}
-				if ($field['field_name'] === 'send_to_sites') {
-					$this->send_to_sites_instrument = $field['form_name'];
-				}
-			}
-			
+	
+            //TODO cache these in separate function
 			// cache event_ids (in order)
 			$this->event_ids = [];
 			foreach ($Proj->eventsForms as $eid => $formList) {
 				$this->event_ids[] = $eid;
 			}
-			
+	
 			// cache list of 2nd event's forms
 			$this->event_2_forms = $Proj->eventsForms[$this->event_ids[1]];
-			
+	
 			// determine name of study intake form instrument
 			$this->study_intake_form_name = 'study_coordinating_center_information';
-			
+	
 			// set label pattern (to convert raw values to label values)
 			$this->label_pattern = "/(\d+),?\s?(.+?)(?=\x{005c}\x{006E}|$)/";
```

Comment:
Consider defining during the definition as opposed to in a method ` private $label_pattern = ...`

Reply:
updated